fuseutil/dirent.go

```diff
@@ -103,3 +117,136 @@ func WriteDirent(buf []byte, d Dirent) (n int) {
 
 	return n
 }
+
+// Write the supplied directory entry with attributes into the given buffer in the format
+// expected in fuseops.ReadDirPlusOp.Dst returning the number of bytes written.
+// Return zero if the entry would not fit.
+func WriteDirentPlus(buf []byte, d DirentPlus) (n int) {
+	type fuse_attr struct {
+		ino       uint64
+		size      uint64
+		blocks    uint64
+		atime     uint64
+		mtime     uint64
+		ctime     uint64
+		atimensec uint32
+		mtimensec uint32
+		ctimensec uint32
+		mode      uint32
+		nlink     uint32
+		uid       uint32
+		gid       uint32
+		rdev      uint32
+		blksize   uint32
+		padding   uint32
+	}
+
+	type fuse_entry_out struct {
+		nodeid           uint64
+		generation       uint64
+		entry_valid      uint64
+		attr_valid       uint64
+		entry_valid_nsec uint32
+		attr_valid_nsec  uint32
+		attr             fuse_attr
+	}
+
+	type fuse_dirent struct {
```

Comment:
Let’s move the fuse_dirent type outside of the function so that we don’t need to duplicate it between WriteDirent and WriteDirentPlus.